<a href="https://colab.research.google.com/github/yumna-09/FlyRank-ML-Internship/blob/main/work/notebooks/w01_research_question.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/yumna-09/FlyRank-ML-Internship/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

In [3]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.



**My lane:** Structured Content Archetype Clustering (Lane 3)
I picked this lane because instead of reviewing hundreds of thousands of pages one at a time, clustering lets me group similar-behaving pages together and understand the content inventory at a glance. This should help identify recurring patterns like pages that are strong, pages that are stale but still visible, and pages that get little attention so decisions can be made about groups rather than individual pages.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

In [4]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


**Decision:** Which content pages a review team should protect, improve, rewrite, or prune first.
**Who acts:** A content strategist or SEO reviewer with limited time each week.
**Cost of a wrong call:** If a page is wrongly labeled "strong" when it's actually declining, it gets ignored and loses more traffic. If a page is wrongly labeled "weak" when it still has value, time gets wasted improving something that didn't need it, and a genuinely valuable page might get pruned by mistake.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [5]:
import os, sys, subprocess

IN_COLAB = "google.colab" in sys.modules
REPO_URL = "https://github.com/yumna-09/FlyRank-ML-Internship"
REPO_DIR = "FlyRank-ML-Internship"

if IN_COLAB:
    if not os.path.isdir(REPO_DIR):
        subprocess.run(["git", "clone", "--depth", "1", REPO_URL, REPO_DIR], check=True)
    os.chdir(REPO_DIR)
else:
    while not os.path.isdir("data/raw") and os.getcwd() != "/":
        os.chdir("..")

print("Working dir:", os.getcwd())
assert os.path.exists("data/raw/content_refresh_anonymized.csv"), "starter CSV not found"

import pandas as pd

df = pd.read_csv("data/raw/content_refresh_anonymized.csv")

print("Distinct content types:", df["content_type"].nunique())
print(df["content_type"].value_counts())

print("\nImpressions_90d spread:")
print(df["impressions_90d"].describe()[["min", "25%", "50%", "75%", "max"]])

print("\nTrend direction counts:")
print(df["trend_direction"].value_counts())

Working dir: /content/FlyRank-ML-Internship/FlyRank-ML-Internship
Distinct content types: 3
content_type
keyword article       27207
feedly article         2096
comparison article      697
Name: count, dtype: int64

Impressions_90d spread:
min         1.00
25%        81.00
50%       731.00
75%      3615.25
max    517715.00
Name: impressions_90d, dtype: float64

Trend direction counts:
trend_direction
down      16262
stable     5962
up         4388
new        2236
flat       1152
Name: count, dtype: int64


Looking at just the starter dataset, three signals already suggest meaningful diversity across content that could form distinct archetypes:

1. Content types vary widely in volume: keyword article (27,207 pages) massively outnumbers feedly article (2,096) and comparison article (697) these types likely behave differently and shouldn't be treated the same way.
2. Impressions_90d has an extreme spread: from 1 to 517,715, with the median at 731 but the 75th percentile at 3,615 this long tail suggests pages fall into very different visibility tiers, not one uniform group.
3. Trend direction is unevenly distributed: 16,262 pages are declining ("down") versus only 4,388 growing ("up"), with meaningful groups also sitting at "stable," "new," and "flat" this alone hints at at least 5 distinct behavioral patterns worth clustering around.

Together, these numbers suggest the content inventory is not homogeneous there's enough variation in type, visibility, and trend to justify grouping pages into archetypes rather than treating them as one pool.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

In [6]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


**What this work can say:**

1. This work can describe observed groupings of content based on measurable signals (impressions, CTR, position, age, word count) not guesses about why those groupings exist.
2. It can offer directional suggestions, like "pages in this cluster tend to have high impressions but low CTR, which may be worth reviewing."
3. It can support decision-making by giving a reviewer a starting point a shortlist of which cluster to look at first not a final answer.

**What this work cannot say:**

1. It cannot claim these clusters reflect Google's actual ranking algorithm or any hidden search-engine logic.
2. It cannot claim a page's cluster membership causes its performance clustering only groups pages that look similar, it doesn't explain cause and effect.
3. It cannot call this "semantic clustering," since the data contains no article text only structured metrics and metadata.
4. It cannot guarantee that acting on a cluster's recommended action (protect/improve/prune) will actually improve outcomes that would require a follow-up experiment, not just clustering.

## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.